praw/models/reddit/subreddit.py

```diff
@@ -2135,18 +2144,78 @@ def create(self, name, content, reason=None, **other_settings):
 
     def revisions(self, **generator_kwargs):
         """Return a generator for recent wiki revisions.
-
         Additional keyword arguments are passed in the initialization of
         :class:`.ListingGenerator`.
-
         To view the wiki revisions for ``'praw_test'`` in ``'/r/test'`` try:
-
         .. code:: python
-
            for item in reddit.subreddit('test').wiki['praw_test'].revisions():
                print(item)
-
         """
         url = API_PATH['wiki_revisions'].format(subreddit=self.subreddit)
         return WikiPage._revision_generator(
             self.subreddit, url, generator_kwargs)
+
+
+class SubredditEmoji(object):
+    """Provides a set of moderation functions to a Subreddit."""
+
+    def __getitem__(self, page_name):
+        """Lazily return the Emoji for the subreddit named ``page_name``.
+
+        This method is to be used to fetch a specific emoji url, like so:
+
+        .. code:: python
+
+           emoji = reddit.subreddit('iama').wiki['cake']
+           print(wikipage.url)
+
+        """
+        return Emoji(self.subreddit._reddit, self.subreddit,
+                        emoji.lower())
+
+    def __init__(self, subreddit):
+        """Create a SubredditModeration instance.
+
+        :param subreddit: The subreddit to moderate.
+
+        """
+        self.subreddit = subreddit
+
+    def __iter__(self):
+        """Iterate through the Emoji for the subreddit.
+
+        This method is to be used to discover all emoji for a subreddit:
+
+        .. code:: python
+
+           for emoji in reddit.subreddit('iama').emoji:
+               print(emoji)
+
+        """
+        response = self.subreddit._reddit.get(
+            API_PATH['emoji_list'].format(subreddit=self.subreddit),
+            params={'unique': self.subreddit._reddit._next_unique})
+        for emoji_name in response[self.subreddit]:
+            yield Emoji(self.subreddit._reddit, self.subreddit, emoji_name)
+
+    def create(self, name, content, reason=None, **other_settings):
+        """Create a new emoji page.
+
+        :param name: The name of the new Emoji. This name will be
+            normalized.
+        :param content: The content of the new Emoji.
+        :param reason: (Optional) The reason for the creation.
+        :param other_settings: Additional keyword arguments to pass.
+
+        To create the emoji page ``'praw_test'`` in ``'/r/test'`` try:
+
+        .. code:: python
+
+           reddit.subreddit('test').emoji.create(
+               'praw_test', 'wiki body text', reason='PRAW Test Creation')
+
+        """
+        name = name.replace(' ', '_').lower()
+        new = Emoji(self.subreddit._reddit, self.subreddit, name)
+        new.edit(content=content, reason=reason, **other_settings)
```

Comment:
It doesn't look like this method works.

Reply:
Yeah I deleted this whole method, since its functionality is actually in emoji.py